reVX/least_cost_xmission/transmission_layer_creator_cli.py

```diff
@@ -0,0 +1,256 @@
+              default='.',
+              help='Directory to store mask GeoTIFFs in.')
+@click.option('--dont-reproject', '-d', is_flag=True, default=False,
+              help='Don\'t reproject vector before creating masks.')
+def create_masks(land_mask_vector: str, template_raster: str, masks_dir: str,
+                 dont_reproject: bool):
+    """
+    Convert a vector land file to wet, dry, and all other masks.
+    """
+    io_handler = TransLayerIoHandler(template_raster)
+    masks = Masks(io_handler, masks_dir=masks_dir)
+
+    reproject = not dont_reproject
+    masks.create_masks(land_mask_vector, save_tiff=True,
+                       reproject_vector=reproject)
+
+
+# @main.command
+@click.option('--template-raster', '-t', type=click.Path(exists=True),
+              required=True,
+              help='Raster to extract CRS, transform, and shape from.')
+@click.option('--h5-file', '-h', type=click.Path(exists=False), required=True,
+              help='Name of H5 file to create.')
+def FUTURE_create_h5(template_raster: str, h5_file: str):
+    """
+    Create a new H5 file to store layers in.
+    """
+    logger.info(
+        f'Using raster {template_raster} to create new H5 file {h5_file}'
+    )
+    ExclusionsConverter._init_h5(h5_file, template_raster)
+
+
+@main.command
+@click.option('--template-raster', '-t', type=click.Path(exists=True),
+              required=True,
+              help='Raster to extract CRS, transform, and shape from.')
+@click.option('--existing-h5-file', '-e', type=click.Path(exists=True),
+              required=True,
+              help='Existing H5 file with correct shape and profile.')
+@click.option('--new-h5-file', '-n', type=click.Path(exists=False),
+              required=True, help='Name of H5 file to create.')
+def create_h5(template_raster: str, existing_h5_file: str,
+              new_h5_file: str):
+    """
+    Create a new H5 file to store layers in.
+    """
+    logger.info(f'Creating new H5 {new_h5_file} with meta data from '
+                f'{existing_h5_file}')
+    io_handler = TransLayerIoHandler(template_raster)
+    io_handler.create_new_h5(existing_h5_file, new_h5_file)
+
+
+def _combine_friction_and_barriers(config: MergeFrictionBarriers,
+                                   io_handler: TransLayerIoHandler,
+                                   save_tiff: bool = True):
+    """
+    Combine friction and barriers and save to H5 and optionally GeoTIFF
+
+    Parameters
+    ----------
+    config : MergeFrictionBarriers
+        Config object
+    io_handler : TransLayerIoHandler
+        Transmission IO handler
+    save_tiff : bool, optional
+        Save combined barriers to GeoTIFF if True. By default, ``True``.
+    """
+    if not Path(FRICTION_TIFF).exists():
+        logger.error(f'The friction GeoTIFF ({FRICTION_TIFF}) was not found. '
+                     'Please create it using the `friction_layers` key in the '
+                     'config file.')
```

Comment:
Does this code need Exceptions raise here? Or can it continue without the files?

Reply:
Added `sys.exit(1)`. My reasoning is that a user facing CLI should throw minimal, if any, exceptions. That concept has not been fully implemented in this code though.